api/v3/Job/Iatsrecurringcontributions.php

```diff
@@ -201,6 +201,18 @@ function civicrm_api3_job_Iatsrecurringcontributions($params) {
       $contribution['skipLineItem'] = 1;
       $contribution['api.line_item.create'] = $contribution_template['line_items'];
     }
+    if (empty($contribution['id'])) {
```

Comment:
@monishdeb is there a reason why this couldn't be done in a pre hook in the canadahelps extension instead of in Iats? are hooks not being properly called or something?

Reply:
@seamuslee001 only reason is to reduce checkings in pre/post hook. If I do the same in pre/post hook I need to check for every contribution with `!empty(contribution_recur_id)` and there's already a similar code for setting custom field value. So I thought if I put the code in this place, it will act on ONLY recurring subsequent payments created at this point.

Reply:
My feeling is that this isn't really code tied to IATS its more about any recurring contribution (which could be IATS / stripe etc) which to me feels like it should live in the main extension thoughts @Edzelopez

Reply:
Yep, I second that @seamuslee001